src/charm.py

```diff
@@ -199,6 +193,19 @@ def __init__(self, *args):
         self.framework.observe(self.on.list_backups_action, self._on_list_backups_action)
         self.framework.observe(self.on.restore_backup_action, self._on_restore_backup_action)
 
+    def generate_vault_scrape_configs(self):
+        """Generate the scrape configs for the COS agent."""
+        return [
+                {
+                    "scheme": "https",
+                    "tls_config": {
+                        "insecure_skip_verify": False,
+                        "ca": self.tls.pull_tls_file_from_workload(File.CA),
+                    },
+                    "metrics_path": "/v1/sys/metrics",
+                    "static_configs": [{"targets": [f"{self._bind_address}:{VAULT_PORT}"]}],
```

Comment:
Before the peer relation is created, I'm pretty sure the bind address will be `None`. In this case, we should probably return an empty list instead of providing a target with the "None:8200" address. I suggest adding a conditional that validates the existence of this address at the beginning of this method, if it's None, then return an empty list.

Reply:
Done, although I don't think it would have cause problems, scraping simply is not going to work until the configuration is complete, grafana checks for that using refresh event, but your comment brought my attention to adding a refresh event of our own, just to be sure, I adde peer-relation-changed event as a refresh event trigger.